Index: src/complete.c
===================================================================
--- src/complete.c (revision 74651f2e9f112eacc07ed9d1281e27ce6a4be9d4)
+++ src/complete.c (revision 2ac7aeadc8560e8830877677192d0c28290c6a33)
@@ -1026,5 +1026,5 @@
                 GList *e;
 
-                for (i = 0, e = ((WListbox *) h->current)->list;
+                for (i = 0, e = ((WListbox *) h->current->data)->list;
                      e != NULL; i++, e = g_list_next (e))
                 {
@@ -1033,5 +1033,5 @@
                     if (strncmp (input->buffer + start, le->text, end - start - 1) == 0)
                     {
-                        listbox_select_entry ((WListbox *) h->current, i);
+                        listbox_select_entry ((WListbox *) h->current->data, i);
                         end = str_get_prev_char (&(input->buffer[end])) - input->buffer;
                         handle_char (input, parm);
@@ -1078,5 +1078,5 @@
                 }
 
-                for (i = 0, e = ((WListbox *) h->current)->list;
+                for (i = 0, e = ((WListbox *) h->current->data)->list;
                      e != NULL; i++, e = g_list_next (e))
                 {
@@ -1118,5 +1118,5 @@
                             {
                                 need_redraw = 1;
-                                listbox_select_entry ((WListbox *) h->current, i);
+                                listbox_select_entry ((WListbox *) h->current->data, i);
                                 last_text = le->text;
                             }
