--- mc-4.8.30/src/editor/edit.c.orig	2023-08-06 15:26:05.000000000 +0200
+++ mc-4.8.30/src/editor/edit.c	2023-08-14 01:05:51.126667829 +0200
@@ -99,8 +99,8 @@
     .backup_ext = NULL,
     .filesize_threshold = NULL,
     .stop_format_chars = NULL,
-    .visible_tabs = TRUE,
-    .visible_tws = TRUE,
+    .visible_tabs = FALSE,
+    .visible_tws = FALSE,
     .show_right_margin = FALSE,
     .simple_statusbar = FALSE,
     .check_nl_at_eof = FALSE
