--- mc-4.8.19/src/editor/editdraw.c.orig	2017-03-04 18:51:38.000000000 +0100
+++ mc-4.8.19/src/editor/editdraw.c	2017-03-11 18:27:31.941897138 +0100
@@ -63,8 +63,8 @@
 /* Toggles statusbar draw style */
 gboolean simple_statusbar = FALSE;
 
-gboolean visible_tws = TRUE;
-gboolean visible_tabs = TRUE;
+gboolean visible_tws = FALSE;
+gboolean visible_tabs = FALSE;
 
 /*** file scope macro definitions ****************************************************************/
 
