--- mc-4.7.5/src/filemanager/midnight.c.orig	2010-12-28 10:30:45.000000000 +0100
+++ mc-4.7.5/src/filemanager/midnight.c	2010-12-29 00:10:55.000000000 +0100
@@ -750,6 +750,9 @@ init_xterm_support (void)
         || strcmp (termvalue, "Eterm") == 0 || strcmp (termvalue, "dtterm") == 0)
     {
         xterm_flag = 1;
+#ifdef HAVE_SLANG
+	tty_display_8bit (0);
+#endif  
 
         /* Default to the standard xterm sequence */
         if (!xmouse_seq)
